### PR TITLE
CV drive FSM integration

### PR DESCRIPTION
**Pull request by AknA13**
Added CV align and drive-up states with the drive FSMs. This allows CV-guided drive to be run during teleop.

### CODE REVIEW

File: src/main/java/frc/robot/Constants.java [hunk 1]
```diff
@@ -27,10 +36,10 @@ public class Constants {
 	public static final double GYRO_TURN_MULTIPLER_BELOW_90 = 0.987;
 	public static final double CHARGING_STATION_LEVELED_ERROR_DEGREES = 2;
 	public static final double CHARGING_STATION_BALANCE_CONSTANT_PID_P = 200;
-	public static final double AUTONOMUS_MOVE_POWER = 0.3;
-	public static final double AUTONOMUS_MOVE_THRESHOLD = 1.5;
+	public static final double AUTONOMUS_MOVE_POWER = 0.05;
+	public static final double AUTONOMUS_MOVE_THRESHOLD = 5;
 
-	// Path points added
+	// Path points
```
Comment (AG6GR) [inline src/main/java/frc/robot/Constants.java:42]: What are the units here? Also I really feel like these points should be [Translation2D](https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/math/geometry/Translation2d.html) objects

File: src/main/java/frc/robot/Constants.java [hunk 2]
```diff
@@ -53,15 +74,24 @@ public class Constants {
 	public static final double HUB_Y_COORDINATE = 0;
 
 	public static final double QUARTER_PI = Math.PI / 4;
-	public static final double ENCODER_TICKS_TO_METERS_CONSTANT = 23.1889;
 
 	static class VisionConstants {
 		static final double CAM_OFFSET_X_METERS = Units.inchesToMeters(0);
-		static final double CAM_OFFSET_Y_METERS = Units.inchesToMeters(0);
-		static final double CAM_HEIGHT_METERS = Units.inchesToMeters(20.00);
+		static final double CAM_OFFSET_Y_METERS = Units.inchesToMeters(8.6);
```
Comment (AG6GR) [inline src/main/java/frc/robot/Constants.java:80]: If we already have a VisionConstants subsection, can we put the rest of the CV related constants under here as well for organization?

File: src/main/java/frc/robot/HardwareMap.java [hunk 3]
```diff
@@ -8,37 +8,29 @@
  */
 public final class HardwareMap {
 	// ID numbers for devices on the CAN bus
-	public static final int CAN_ID_SPARK_PIVOT = 5;
-	public static final int CAN_ID_SPARK_TELEARM = 10;
+	public static final int CAN_ID_SPARK_PIVOT = 32;
+	public static final int CAN_ID_SPARK_TELEARM = 5;
+	// public static final int CAN_ID_SPARK_DRIVE_LEFT1 = 31;
```
Comment (AG6GR) [inline src/main/java/frc/robot/HardwareMap.java:13]: Please rebase for the latest SPARK Max ID assignments.

File: src/main/java/frc/robot/systems/ArmFSM.java [hunk 4]
```diff
@@ -50,7 +49,7 @@ public enum FSMState {
 	private static final double BALANCE_ANGLE_ENCODER_ROTATIONS = 5;
 	private static final double GRABBER_ANGLE_ENCODER_ROTATIONS = -5;
 	private static final double ARM_ENCODER_GRAB_ROTATIONS = 10;
-	private static final double PID_MAX_POWER = 0.2;
+	private static final double PID_MAX_POWER = 0.3;
```
Comment (AG6GR) [inline src/main/java/frc/robot/systems/ArmFSM.java:52]: Please reset the FSM you didn't modify to the version on main.

File: src/main/java/frc/robot/systems/DriveFSMSystem.java [hunk 5]
```diff
@@ -29,11 +25,15 @@ public class DriveFSMSystem {
 	public enum FSMState {
 		TELE_STATE_2_MOTOR_DRIVE,
 		TELE_STATE_BALANCE,
-		TELE_STATE_CV_ALIGN,
 		TELE_STATE_MECANUM,
 		PURE_PURSUIT,
 		TURNING_STATE,
+		CVLowTapeAlign,
```
Comment (AG6GR) [inline src/main/java/frc/robot/systems/DriveFSMSystem.java:31]: Constants should be all caps.

File: src/main/java/frc/robot/systems/DriveFSMSystem.java [hunk 6]
```diff
@@ -46,14 +46,28 @@ public enum FSMState {
...
+		P4N7,
+		P4N8,
 	}
 
 	/* ======================== Private variables ======================== */
 	private FSMState currentState;
 
 	// Hardware devices should be owned by one and only one system. They must
 	// be private to their owner system and may not be used elsewhere.
+	// private CANSparkMax leftMotor1;
```
Comment (AG6GR) [inline src/main/java/frc/robot/systems/DriveFSMSystem.java:66]: Clean up unused code

File: src/main/java/frc/robot/systems/DriveFSMSystem.java [hunk 7]
```diff
@@ -70,15 +84,15 @@ public enum FSMState {
 	private double prevEncoderPos = 0;
 	private double gyroAngleForOdo = 0;
 	private AHRS gyro;
-	private double startAngle;
-
-	private double angleToTurnToFaceTag = 0;
 
-	private DrivePoseEstimator dpe = new DrivePoseEstimator();
+	private boolean forward = false;
```
Comment (AG6GR) [inline src/main/java/frc/robot/systems/DriveFSMSystem.java:88]: Please use a more descriptive name. Also booleans should all be named starting with "is" i.e. "isXYZ"

File: src/main/java/frc/robot/systems/DriveFSMSystem.java [hunk 8]
```diff
@@ -89,22 +103,46 @@ public enum FSMState {
 	 */
 	public DriveFSMSystem() {
 		// Perform hardware init
+		// leftMotor1 = new CANSparkMax(HardwareMap.CAN_ID_SPARK_DRIVE_LEFT1,
```
Comment (AG6GR) [inline src/main/java/frc/robot/systems/DriveFSMSystem.java:106]: Remove unused code.

File: src/main/java/frc/robot/systems/DriveFSMSystem.java [hunk 9]
```diff
@@ -460,32 +550,42 @@ private void handleTeleOp2MotorState(TeleopInput input) {
...
 			rightPower = power.getRightPower();
 
-			if (!pcw.getEstimatedGlobalPose().isEmpty()) {
-				// left is negative right is positive
-				angleToTurnToFaceTag = Math.abs((Constants.HALF_REVOLUTION_DEGREES
-					+ Constants.ONE_REVOLUTION_DEGREES - Math.toDegrees(
-					cvEstimatedPos.getRotation().getAngle())) + Math.toDegrees(
-						Math.atan2(cvEstimatedPos.getY(),
-						cvEstimatedPos.getX())));
+			// if (!pcw.getEstimatedGlobalPose().isEmpty()) {
```
Comment (AG6GR) [inline src/main/java/frc/robot/systems/DriveFSMSystem.java:553]: Remove unused code

**Review by AG6GR — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/Constants.java (hunk 1 above)
Comment (AknA13) [reply src/main/java/frc/robot/Constants.java:42]: That's up to drive on whether they want to use Translation2d or represent their points as these.

Comment (SamNat975) [reply src/main/java/frc/robot/Constants.java:32]: Drive said they preferred their points like this so we're just leaving them like that.

**Comment by SamNat975** [comment]
i rebased off of main and i had to squash all the commits on this into one commit— and then mavis told me to force-push the dummy branch onto this pull request
the code still needs to be fixed because it has errors in teleop though

**Comment by SamNat975** [comment]
addressed merge conflict errors

**Review by AG6GR — APPROVED** [review]

**Pull request merged**